core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -131,7 +131,10 @@
     private final Map<NodeRef<Expression>, ResolvedField> columnReferences = new LinkedHashMap<>();
 
     // a map of users to the columns per table that they access
-    private final Map<AccessControlInfo, Map<QualifiedObjectName, Set<String>>> tableColumnReferences = new LinkedHashMap<>();
+    private final Map<AccessControlInfo, Map<QualifiedObjectName, Set<String>>> tableColumnReferencesInStatements = new LinkedHashMap<>();
+
+    // a map of users to the columns per table that they access in row filter
```

Comment:
Row filter could be one use case. I think it should be more generic to mention that it is before filtering so it contains all columns (including restricted).

So I think one should be called `tableColumnReferences` and other `filteredTableColumnReferences`. Or even `columnReferences` and other `filteredColumnReferences`.

Reply:
I'll use filteredTableColumnReferences and directTableColumnReferences so that i can have getter with the usually naming convention: getTableColumnReferences already exists and for retrocompatibily I could not change that signature, is it ok?